src/SfdxProjectBuilder.groovy

```diff
@@ -857,89 +857,117 @@ class SfdxProjectBuilder implements Serializable {
+          _.error "hub org authorization failed" 
+        }
     }
   }
 
   private void createScratchOrg() {
-    _.echo('Creating scratch org')
-
-    def commandScriptString = "sfdx org create scratch --definition-file ${this.sfdxScratchOrgDefinitionFile} --json --duration-days 1 --alias ${this.sfdxScratchOrgAlias} --target-dev-hub ${_.env.SFDX_DEV_HUB_USERNAME} --wait 30"
-
-    def response
-
-    try {
-      debug('before call to sh command to create org')
-      def rmsg = _.sh returnStdout: true, script: commandScriptString
-      debug('after the call to sh command to create org')
-      debug(rmsg)
-      response = jsonParse( rmsg )
-      debug('after the parsing of rmsg')
-      debug('------------------------------------------------------')
-      debug('response == ')
-      debug(response)
-      debug('------------------------------------------------------')
-    }
-    catch (ex) {
-      // printf ex
-      _.echo('------------------------------------------------------')
-      // _.echo(ex)
-      // _.echo(ex.status)
-      // _.echo(ex.name)
-      // _.echo(ex.message)
-      _.echo('------------------------------------------------------')
-      // if (ex.getMessage().contains('OPERATION_TOO_LARGE')) {
-      if (ex.message.contains('OPERATION_TOO_LARGE')) {
-        _.echo('exception message contains OPERATION_TOO_LARGE')
-        _.echo('------------------------------------------------------')
-        _.echo(ex.printStackTrace())
-        _.error "Failed to create Scratch Org -- ${response.message}"
-      } else if (ex.name.equals('genericTimeoutMessage') || ex.name.equals('RemoteOrgSignupFailed')) {
-        // try one more time to create the scratch org
-        _.echo('Original attempt to create scratch org timed out.  Trying to create one again.')
-        rmsg = _.sh returnStdout: true, script: commandScriptString
-        response = jsonParse( rmsg )
-        if ( response.status != 0 ) {
-          _.error "Failed to create Scratch Org -- ${response.message}"
-        }
+      _.echo('Creating scratch org')
+  
+      def commandScriptString = "sfdx org create scratch --definition-file ${this.sfdxScratchOrgDefinitionFile} --json --duration-days 1 --alias ${this.sfdxScratchOrgAlias} --target-dev-hub ${_.env.SFDX_DEV_HUB_USERNAME} --wait 30"
+  
+      def response
+  
+      try {
+          debug('before call to sh command to create org')
+          def rmsg = _.sh returnStdout: true, script: commandScriptString
+          debug('after the call to sh command to create org')
+          debug(rmsg)
+          response = jsonParse(rmsg)
+          debug('after the parsing of rmsg')
+          debug('------------------------------------------------------')
+          debug('response == ')
+          debug(response)
+          debug('------------------------------------------------------')
+  
+          // Push source code to the scratch org
```

Comment:
@FernandoQuezado -- I am a bit confused here.  Why are you trying to push code to the scratch during the `createScratchOrg()` method?  Doing that would pull the entire deployment of source code forward to the initializaiton stage and bypass the validateStage and processResourcesStages completely.